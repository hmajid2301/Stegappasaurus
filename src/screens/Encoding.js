import React, { Component } from 'react';
import { View, Text } from 'react-native';

import styles from '../components/Common/colors';


export default class Encoding extends Component {
  render() {
    return (
      <View style={styles.showCanvas}>
        <Text> Encoding </Text>
      </View>
    );
  }
}
