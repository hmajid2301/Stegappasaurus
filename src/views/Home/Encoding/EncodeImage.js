import { FileSystem, MediaLibrary } from 'expo';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import {
  Linking,
  Platform,
} from 'react-native';

import ImageMessage from '~/views/Home/ImageMessage';
import ImageProgressCircle from '~/views/Home/ImageProgressCircle';
import { colors } from '~/util/styles';


export default class EncodeImage extends Component {
  constructor(props) {
    const { navigation } = props;
    const uri = navigation.getParam('uri', 'NO-ID');

    super(props);
    this.state = {
      isEncoding: false,
      photo: uri,
    };
  }

  static propTypes = {
    navigation: PropTypes.object.isRequired,
    screenProps: PropTypes.shape({
      theme: PropTypes.shape({
        isDark: PropTypes.bool.isRequired,
        background: PropTypes.string.isRequired,
        color: PropTypes.string.isRequired,
      }),
    }),
  }

  isEncoding = () => {
    this.setState({ isEncoding: true });
  }

  encoded = async () => {
    await FileSystem.writeAsStringAsync(`${FileSystem.documentDirectory}tmpimg.jpg`, 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAMAAABEpIrGAAAAA3NCSVQICAjb4U/gAAAACXBIWXMAAAPzAAAD8wF1XGupAAAAGXRFWHRTb2Z0d2FyZQB3d3cuaW5rc2NhcGUub3Jnm+48GgAAAJZQTFRF////SUmSQGCfQFWVPVyZOVuZO1eaPFmZO1qYO1mZO1mYO1mYOlmYPFmYO1mYO1mYO1qYO1mYPVuZP1yaQV6bQ2CcR2OeTWmiVG6lVW+mVnCmV3GnWnOoXnerZHytZn2va4Kxcoe1eY65gJS8ipzCi53CkKHFtsHZuMPau8bbwsvf09ro1Nvo1dvp4ubw9vf6/Pz9////XyoQ3AAAABF0Uk5TAAcIGBktSYSXmMHI2uPy8/XVqDFbAAABA0lEQVQ4y4WT2WKDIBBFcYkswbVp9n2pra1N/P+fC5gII5B4n8B7wJlhBiElL6KMpylnNPKQrZAkuVJCQsP2cZb3lGEf+sE4tzQOtD+Kc4fikTrv9AXxvMMH90+/vn/r+tj95REH1v78v5E6d3vc5gfi/2n95qJykdkS7X/chHut/47qCxH1A/VZyOMHGGfioQhs1xJY9zKJEFXrYrqVwGYyKTRAEVPrXdPppAGGuAPYa4Cj1AGsNJACYFlW0q3K8hMC/H0WHATpBBhI0wnQ4ULBUtuAKDV8LBsg/ee2gPa5QcNYADZazgSeLaeb1gDiwGz7YiZU2G0/PDjDozc8vK/H/w603kSHess3kQAAAABJRU5ErkJggg==').then(
      () => { console.log('Yes it is fin!'); });
    await MediaLibrary.createAssetAsync(`${FileSystem.documentDirectory}tmpimg.jpg`);
  }

  openPhotos = () => {
    switch (Platform.OS) {
      case 'ios':
        Linking.openURL('photos-redirect://');
        break;
      case 'android':
        Linking.openURL('content://media/internal/images/media');
        break;
      default:
        throw new Error('Could not open gallery app');
    }
  }

  render() {
    const { theme } = this.props.screenProps;
    if (this.state.isEncoding) {
      return (
        <ImageProgressCircle
          action={this.encoded}
          message={'Saved Encoded Photo'}
          snackAction={this.openPhotos}
          photo={this.state.photo}
          primaryColor={colors.primary}
          theme={theme}
        />
      );
    }
    return (
      <ImageMessage action={this.isEncoding} photo={this.state.photo}/>
    );
  }
}
