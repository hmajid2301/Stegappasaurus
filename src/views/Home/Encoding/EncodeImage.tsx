import { FileSystem, MediaLibrary } from "expo";
import React, { Component } from "react";
import { View } from "react-native";
import Canvas, { Image as CanvasImage, ImageData } from "react-native-canvas";
import { NavigationScreenProp } from "react-navigation";

import ImageMessage from "~/components/ImageMessage";
import ImageProgressCircle from "~/components/ImageProgressCircle";
import { withDispatchAlgorithm } from "~/redux/hoc";
import Steganography from "~/services/steganography";
import { AlgorithmNames, ITheme, PrimaryColor } from "~/util/interfaces";
import { colors } from "~/util/styles";

interface IProps {
  algorithm: AlgorithmNames;
  navigation: NavigationScreenProp<any, any>;
  screenProps: {
    theme: ITheme;
  };
}

interface IState {
  isEncoding: boolean;
  message: string;
  photo: string;
}

class EncodeImage extends Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    const { navigation } = props;
    const uri = navigation.getParam("uri", "NO-ID");

    this.state = {
      isEncoding: false,
      message: "",
      photo: uri
    };
  }

  public render() {
    const { theme } = this.props.screenProps;
    if (this.state.isEncoding) {
      return (
        <View>
          <ImageProgressCircle
            action={this.encoded}
            message={"Saved Encoded Photo"}
            photo={this.state.photo}
            primaryColor={colors.primary as PrimaryColor}
            theme={theme}
          />
          <Canvas ref={this.encodeData} />
        </View>
      );
    }
    return <ImageMessage action={this.isEncoding} photo={this.state.photo} />;
  }

  private isEncoding = (message: string) => {
    this.setState({ isEncoding: true, message });
  };

  private encoded = async () => {
    await FileSystem.writeAsStringAsync(
      `${FileSystem.documentDirectory}tmpimg.jpg`,
      "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAMAAABEpIrGAAAAA3NCSVQICAjb4U/gAAAACXBIWXMAAAPzAAAD8wF1XGupAAAAGXRFWHRTb2Z0d2FyZQB3d3cuaW5rc2NhcGUub3Jnm+48GgAAAJZQTFRF////SUmSQGCfQFWVPVyZOVuZO1eaPFmZO1qYO1mZO1mYO1mYOlmYPFmYO1mYO1mYO1qYO1mYPVuZP1yaQV6bQ2CcR2OeTWmiVG6lVW+mVnCmV3GnWnOoXnerZHytZn2va4Kxcoe1eY65gJS8ipzCi53CkKHFtsHZuMPau8bbwsvf09ro1Nvo1dvp4ubw9vf6/Pz9////XyoQ3AAAABF0Uk5TAAcIGBktSYSXmMHI2uPy8/XVqDFbAAABA0lEQVQ4y4WT2WKDIBBFcYkswbVp9n2pra1N/P+fC5gII5B4n8B7wJlhBiElL6KMpylnNPKQrZAkuVJCQsP2cZb3lGEf+sE4tzQOtD+Kc4fikTrv9AXxvMMH90+/vn/r+tj95REH1v78v5E6d3vc5gfi/2n95qJykdkS7X/chHut/47qCxH1A/VZyOMHGGfioQhs1xJY9zKJEFXrYrqVwGYyKTRAEVPrXdPppAGGuAPYa4Cj1AGsNJACYFlW0q3K8hMC/H0WHATpBBhI0wnQ4ULBUtuAKDV8LBsg/ee2gPa5QcNYADZazgSeLaeb1gDiwGz7YiZU2G0/PDjDozc8vK/H/w603kSHess3kQAAAABJRU5ErkJggg=="
    ).then();
    await MediaLibrary.createAssetAsync(
      `${FileSystem.documentDirectory}tmpimg.jpg`
    );
  };

  private encodeData = async (canvas: Canvas) => {
    const image = new CanvasImage(canvas);
    image.addEventListener("load", () => {
      context.drawImage(image, 0, 0);
    });
    image.src = this.state.photo;
    const context = canvas.getContext("2d");
    const imageData = await context.getImageData(
      0,
      0,
      image.width,
      image.height
    );
    const steganography = new Steganography(this.props.algorithm, imageData);
    const encodedData = steganography.encode(this.state.message);

    const imgData = new ImageData(
      canvas,
      encodedData,
      image.width,
      image.height
    );
    context.putImageData(imgData, 0, 0);
    return canvas.toDataURL();
  };
}

export default withDispatchAlgorithm(EncodeImage);
