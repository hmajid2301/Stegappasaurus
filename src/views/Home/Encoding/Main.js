import { ImagePicker, MediaLibrary, Permissions } from 'expo';
import { Icon } from 'native-base';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import {
  ActivityIndicator,
  FlatList,
  Image,
  TouchableOpacity,
  View,
} from 'react-native';
import { connect } from 'react-redux';

import { togglePrimaryColor } from '~/redux/actions';
import { PRIMARY_COLORS } from '~/util/constants';
import { colors } from '~/util/styles';

import styles from './Main/styles';


class Main extends Component {
  constructor(props) {
    super(props);
    this.state = {
      photos: [],
      loading: false,
      lastPhoto: 0,
    };
  }

  static propTypes = {
    navigation: PropTypes.object.isRequired,
    screenProps: PropTypes.shape({
      theme: PropTypes.shape({
        isDark: PropTypes.bool.isRequired,
        background: PropTypes.string.isRequired,
        color: PropTypes.string.isRequired,
      }),
    }),
    togglePrimaryColor: PropTypes.func.isRequired,
  }

  componentDidMount = async () => {
    this.props.navigation.addListener('willFocus', () => {
      this.props.togglePrimaryColor(PRIMARY_COLORS.ORANGE.name);
    });

    const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
    if (status === 'granted') {
      const { assets, endCursor } = await MediaLibrary.getAssetsAsync({
        first: 15,
        sortBy: [MediaLibrary.SortBy.creationTime],
      });
      this.setState({ photos: assets, lastPhoto: endCursor });
    } else {
      throw new Error('Camera Roll permission not granted');
    }
  }

  padData = (data) => {
    const numOfCols = 3;
    const fullRows = Math.floor(data.length / numOfCols);
    let numElementsInFinalRow = data.length - (fullRows * numOfCols);

    while (numElementsInFinalRow !== numOfCols && numElementsInFinalRow !== 0) {
      data.push({ empty: true });
      numElementsInFinalRow += 1;
    }

    return data;
  };

  getPhotoFromCamera = async () => {
    const { status } = await Permissions.askAsync(Permissions.CAMERA, Permissions.CAMERA_ROLL);
    if (status === 'granted') {
      const result = await ImagePicker.launchCameraAsync();
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } else {
      throw new Error('Camera permission not granted');
    }
  }

  getPhotoFromCameraRoll = async () => {
    const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
    if (status === 'granted') {
      const result = await ImagePicker.launchImageLibraryAsync();
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } else {
      throw new Error('Camera Roll permission not granted');
    }
  }

  getPhotoFromCatAPI = () => {
    this.setState({ loading: true });

    const url = 'https://api.thecatapi.com/v1/images/search?mime_types=jpg,png';
    fetch(url)
      .then(response => response.json())
      .then((responseJson) => {
        this.selectPhotoToEncode(responseJson[0].url);
      })
      .catch();

    setTimeout(() => {
      this.setState({ loading: false });
    }, 3000);
  }

  getMorePhotosFromCameraRoll = async () => {
    const { assets, endCursor } = await MediaLibrary.getAssetsAsync({
      first: 9,
      sortBy: [MediaLibrary.SortBy.creationTime],
      after: this.state.lastPhoto,
    });

    this.setState({
      photos: [...this.state.photos, ...assets],
      photoEnd: endCursor,
    });
  }

  renderPhotosFromCameraRoll = ({ item }) => {
    if (item.empty === true) {
      return <View/>;
    }
    return (
      <TouchableOpacity
        onPress={() => this.selectPhotoToEncode(item.uri)}
        style={styles.photoButton}
      >
        <Image source={{ uri: item.uri }} style={styles.photos}/>
      </TouchableOpacity>
    );
  }

  selectPhotoToEncode = (uri) => {
    this.props.navigation.navigate('EncodeImage', { uri });
  }

  render() {
    const { theme } = this.props.screenProps;

    if (this.state.loading) {
      return (
        <View style={styles.loading}>
          <ActivityIndicator color={colors.primary} size='large'/>
        </View>
      );
    }

    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <View style={styles.buttonsRow}>
          <TouchableOpacity onPress={this.getPhotoFromCamera} style={styles.button}>
            <Icon name='camera' style={styles.icon} type='FontAwesome' />
          </TouchableOpacity>
          <TouchableOpacity onPress={this.getPhotoFromCameraRoll} style={styles.button}>
            <Icon name='photo' style={styles.icon} type='FontAwesome'/>
          </TouchableOpacity>
          <TouchableOpacity onPress={this.getPhotoFromCatAPI} style={styles.button}>
            <Icon name='cat' style={styles.icon} type='MaterialCommunityIcons'/>
          </TouchableOpacity>
        </View>
        <View style={styles.photoListContainer}>
          <FlatList
            data={this.padData(this.state.photos)}
            keyExtractor={(_, index) => index}
            numColumns={3}
            onEndReached={this.getMorePhotosFromCameraRoll}
            renderItem={this.renderPhotosFromCameraRoll}
          />
        </View>
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  togglePrimaryColor: color => dispatch(togglePrimaryColor(color)),
});

export default connect(null, mapDispatchToProps)(Main);
