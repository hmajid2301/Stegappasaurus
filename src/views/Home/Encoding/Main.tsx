import { create } from "apisauce";
import { ImagePicker, MediaLibrary, Permissions } from "expo";
import { Icon } from "native-base";
import React, { Component } from "react";
import {
  ActivityIndicator,
  Alert,
  FlatList,
  Image,
  TouchableOpacity,
  View
} from "react-native";
import env from "react-native-dotenv";
import { NavigationScreenProp } from "react-navigation";
import { connect } from "react-redux";
import { Dispatch } from "redux";

import { PRIMARY_COLORS } from "~/common/constants";
import { ITheme, PrimaryColorNames } from "~/common/interfaces";
import { colors } from "~/common/styles";
import { togglePrimaryColor } from "~/redux/actions";

import styles from "./Main/styles";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  screenProps: {
    theme: ITheme;
  };
  togglePrimaryColor: (primaryColor: PrimaryColorNames) => void;
}

interface IState {
  photos: IPhoto[];
  loading: boolean;
  lastPhoto: number;
}

interface IPhoto {
  uri: string;
}

interface ICatAPI {
  breeds: string[];
  id: string;
  url: string;
  width: number;
  height: number;
}

class Main extends Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    this.state = {
      lastPhoto: 0,
      loading: false,
      photos: []
    };
  }

  public render() {
    const { theme } = this.props.screenProps;

    if (this.state.loading) {
      return (
        <View style={styles.loading}>
          <ActivityIndicator color={colors.primary} size="large" />
        </View>
      );
    }

    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <View style={styles.buttonsRow}>
          <TouchableOpacity
            onPress={this.getPhotoFromCamera}
            style={styles.button}
          >
            <Icon name="camera" style={styles.icon} type="FontAwesome" />
          </TouchableOpacity>

          <TouchableOpacity
            onPress={this.getPhotoFromCameraRoll}
            style={styles.button}
          >
            <Icon name="photo" style={styles.icon} type="FontAwesome" />
          </TouchableOpacity>

          <TouchableOpacity
            onPress={this.getPhotoFromCatAPI}
            style={styles.button}
          >
            <Icon
              name="cat"
              style={styles.icon}
              type="MaterialCommunityIcons"
            />
          </TouchableOpacity>
        </View>
        <View style={styles.photoListContainer}>
          <FlatList
            data={this.padData(this.state.photos)}
            keyExtractor={(item, index) => item.uri + index}
            numColumns={3}
            onEndReached={this.morePhotosFromCameraRoll}
            renderItem={this.renderPhotosFromCameraRoll}
          />
        </View>
      </View>
    );
  }

  public componentDidMount = async () => {
    this.props.navigation.addListener("willFocus", () => {
      this.props.togglePrimaryColor(PRIMARY_COLORS.ORANGE.name);
    });

    const { status } = await Permissions.askAsync(
      Permissions.CAMERA,
      Permissions.CAMERA_ROLL
    );
    if (status === "granted") {
      const { assets, endCursor } = await MediaLibrary.getAssetsAsync({
        first: 15,
        sortBy: [MediaLibrary.SortBy.creationTime]
      });
      this.setState({ photos: assets, lastPhoto: endCursor });
    } else {
      Alert.alert(
        "Permissions",
        "Please grant permission to access your camera roll.",
        [
          {
            text: "ok"
          }
        ]
      );
    }
  };

  private padData = (data: IPhoto[]) => {
    const numOfCols = 3;
    const fullRows = Math.floor(data.length / numOfCols);
    let numElementsInFinalRow = data.length - fullRows * numOfCols;

    while (numElementsInFinalRow !== numOfCols && numElementsInFinalRow !== 0) {
      data.push({ uri: "" });
      numElementsInFinalRow += 1;
    }

    return data;
  };

  private getPhotoFromCamera = async () => {
    try {
      const result = await ImagePicker.launchCameraAsync();
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } catch {
      Alert.alert(
        "Permissions",
        "Please grant permission to access your camera.",
        [
          {
            text: "ok"
          }
        ]
      );
    }
  };

  private getPhotoFromCameraRoll = async () => {
    try {
      const result = await ImagePicker.launchImageLibraryAsync();
      if (!result.cancelled) {
        this.selectPhotoToEncode(result.uri);
      }
    } catch {
      Alert.alert(
        "Permissions",
        "Please grant permission to access your camera roll.",
        [
          {
            text: "ok"
          }
        ]
      );
    }
  };

  private getPhotoFromCatAPI = async () => {
    this.setState({ loading: true });

    const api = create({
      baseURL: "https://api.thecatapi.com",
      headers: { "x-api-key": env.CAT_API_KEY }
    });

    const response = await api.get("/v1/images/search?mime_types=jpg,png");
    if (response.ok) {
      const urls = response.data as ICatAPI[];
      this.selectPhotoToEncode(urls[0].url);
    } else {
      Alert.alert(
        "Failed",
        "Failed to get a cat photo, check you're connected to the internet.",
        [
          {
            text: "ok"
          }
        ]
      );
    }

    setTimeout(() => {
      this.setState({ loading: false });
    }, 1000);
  };

  private morePhotosFromCameraRoll = async () => {
    const { assets } = await MediaLibrary.getAssetsAsync({
      after: this.state.lastPhoto,
      first: 9,
      sortBy: [MediaLibrary.SortBy.creationTime]
    });

    this.setState({
      photos: [...this.state.photos, ...assets]
    });
  };

  private renderPhotosFromCameraRoll = ({ item }: { item: IPhoto }) => {
    if (item.uri === "") {
      return <View />;
    }
    return (
      <TouchableOpacity
        onPress={() => this.selectPhotoToEncode(item.uri)}
        style={styles.photoButton}
      >
        <Image source={{ uri: item.uri }} style={styles.photos} />
      </TouchableOpacity>
    );
  };

  private selectPhotoToEncode = (uri: string) => {
    this.props.navigation.navigate("Message", { uri });
  };
}

const mapDispatchToProps = (dispatch: Dispatch) => ({
  togglePrimaryColor: (colorName: PrimaryColorNames) =>
    dispatch(togglePrimaryColor({ color: colorName }))
});

export default connect(
  null,
  mapDispatchToProps
)(Main);
