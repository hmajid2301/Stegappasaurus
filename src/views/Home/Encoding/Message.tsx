import { Toast } from "native-base";
import React, { Component } from "react";
import { Image } from "react-native";
import { NavigationScreenProp } from "react-navigation";

import ImageMessage from "~/components/ImageMessage";
import Steganography from "~/services/steganography";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
}

interface IState {
  photo: string;
}

export default class Message extends Component<IProps, IState> {
  constructor(props: IProps) {
    super(props);
    const { navigation } = props;
    const uri = navigation.getParam("uri", "NO-ID");

    this.state = {
      photo: uri
    };
  }

  public render() {
    return <ImageMessage action={this.onSubmit} photo={this.state.photo} />;
  }

  private onSubmit = (message: string) => {
    let bitsCanEncode = 0;
    Image.getSize(
      this.state.photo,
      (width, height) => {
        bitsCanEncode = (width * height * 3) / 4;
      },
      () => null
    );

    const binaryMessage = new Steganography(
      "LSB-PNG",
      []
    ).convertMessageToBinary(message);
    const messageLength = binaryMessage.join("").length;

    if (messageLength > bitsCanEncode) {
      Toast.show({
        buttonText: "Okay",
        duration: 5000,
        text: "Message too large to encode in image.",
        type: "danger"
      });
    } else {
      this.props.navigation.navigate("Progress", {
        message,
        uri: this.state.photo
      });
    }
  };
}
