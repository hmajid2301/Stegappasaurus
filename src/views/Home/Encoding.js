import { createStackNavigator } from 'react-navigation';

import EncodeImage from './Encoding/EncodeImage';
import Main from './Encoding/Main';


const EncodeNavigator = createStackNavigator({
  Main: {
    screen: Main,
    navigationOptions: {
      header: null,
    },
  },

  EncodeImage: {
    screen: EncodeImage,
    navigationOptions: {
      header: null,
      tabBarVisible: false,
    },
  },
});

EncodeNavigator.navigationOptions = ({ navigation }) => ({
  tabBarVisible: navigation.state.index === 0,
  swipeEnabled: navigation.state.index === 0,
});

export default EncodeNavigator;
