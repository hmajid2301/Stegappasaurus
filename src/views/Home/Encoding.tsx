import {
  createStackNavigator,
  NavigationRoute,
  NavigationScreenProp
} from "react-navigation";

import Main from "./Encoding/Main";
import Message from "./Encoding/Message";
import Progress from "./Encoding/Progress";

const EncodeNavigator = createStackNavigator({
  Main: {
    navigationOptions: {
      header: null
    },
    screen: Main
  },
  Message: {
    navigationOptions: {
      header: null,
      tabBarVisible: false
    },
    screen: Message
  },
  Progress: {
    navigationOptions: {
      header: null,
      tabBarVisible: false
    },
    screen: Progress
  }
});

EncodeNavigator.navigationOptions = ({
  navigation
}: {
  navigation: NavigationScreenProp<NavigationRoute>;
}) => ({
  swipeEnabled: navigation.state.index === 0,
  tabBarVisible: navigation.state.index === 0
});

export default EncodeNavigator;
