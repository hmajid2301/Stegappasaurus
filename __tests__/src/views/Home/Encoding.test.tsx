import { shallow } from "enzyme";
import React from "react";

import Encoding from "~/views/Home/Encoding";

describe("Encoding: Match Snapshots", () => {
  test("1", () => {
    const component = shallow(<Encoding />);
    expect(component).toMatchSnapshot();
  });
});
