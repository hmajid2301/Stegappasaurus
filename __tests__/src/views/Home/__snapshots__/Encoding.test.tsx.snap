// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Encoding: Match Snapshots 1 1`] = `
<Navigator
  onGestureBegin={[Function]}
  onGestureCanceled={[Function]}
  onGestureEnd={[Function]}
  onTransitionStart={[Function]}
/>
`;
